StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/MaintainabilityRules/SA1412FixAllProvider.cs

```diff
@@ -3,77 +3,111 @@
 
 namespace StyleCop.Analyzers.MaintainabilityRules
 {
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Threading;
     using System.Threading.Tasks;
+    using Helpers;
     using Microsoft.CodeAnalysis;
     using Microsoft.CodeAnalysis.CodeActions;
     using Microsoft.CodeAnalysis.CodeFixes;
 
     internal sealed class SA1412FixAllProvider : FixAllProvider
     {
-        public override async Task<CodeAction> GetFixAsync(FixAllContext fixAllContext)
+        public override Task<CodeAction> GetFixAsync(FixAllContext fixAllContext)
         {
-            Solution newSolution;
+            string title = string.Format(MaintainabilityResources.SA1412CodeFix, fixAllContext.CodeActionEquivalenceKey.Substring(fixAllContext.CodeActionEquivalenceKey.IndexOf('.') + 1));
 
+            CodeAction fixAction;
             switch (fixAllContext.Scope)
             {
             case FixAllScope.Document:
-                newSolution = await FixDocumentAsync(fixAllContext, fixAllContext.Document).ConfigureAwait(false);
+                fixAction = CodeAction.Create(
+                    title,
+                    cancellationToken => this.GetDocumentFixesAsync(fixAllContext.WithCancellationToken(cancellationToken)),
+                    nameof(SA1412FixAllProvider));
                 break;
 
             case FixAllScope.Project:
-                newSolution = await GetProjectFixesAsync(fixAllContext, fixAllContext.Project).ConfigureAwait(false);
+                fixAction = CodeAction.Create(
+                    title,
+                    cancellationToken => this.GetProjectFixesAsync(fixAllContext.WithCancellationToken(cancellationToken), fixAllContext.Project),
```

Comment:
:question: Why does this need to pass the project? Its already in the `fixAllContext` as far as I can determine.